drivers/unix/file_access_unix.cpp

```diff
@@ -117,7 +117,29 @@ Error FileAccessUnix::open_internal(const String &p_path, int p_mode_flags) {
 #endif
 
 	if (is_backup_save_enabled() && (p_mode_flags == WRITE)) {
-		save_path = path;
+		// Set save path to the symlink target, not the link itself.
+		String link;
+		bool is_link = false;
+		{
+			CharString cs = path.utf8();
+			struct stat lst = {};
+			if ((lstat(cs.get_data(), &lst) == 0)) {
+				is_link = S_ISLNK(lst.st_mode);
+			}
+			if (is_link) {
+				char buf[256];
```

Comment:
Why 256? Compare https://man7.org/linux/man-pages/man2/readlink.2.html#EXAMPLES

Reply:
Same as all other uses of `readlink`, but I guess `PATH_MAX` make more sense.